MQ2AutoSize.cpp

```diff
@@ -1,627 +1,1387 @@
+		ResizeSelf,
+		Range,
+		SizePC,
+		SizeNPC,
+		SizePets,
+		SizeMercs,
+		SizeMounts,
+		SizeCorpse,
+		SizeSelf
+	};
+
+	MQ2AutoSizeType() :MQ2Type("AutoSize") {
+		TypeMember(Active);
+		TypeMember(AutoSave);
+		TypeMember(ResizePC);
+		TypeMember(ResizeNPC);
+		TypeMember(ResizePets);
+		TypeMember(ResizeMercs);
+		TypeMember(ResizeMounts);
+		TypeMember(ResizeCorpse);
+		TypeMember(ResizeSelf);
+		TypeMember(Range);
+		TypeMember(SizePC);
+		TypeMember(SizeNPC);
+		TypeMember(SizePets);
+		TypeMember(SizeMercs);
+		TypeMember(SizeMounts);
+		TypeMember(SizeCorpse);
+		TypeMember(SizeSelf);
+	}
+
+	~MQ2AutoSizeType() {}
+
+	bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) {
+
+		MQTypeMember* pMember = MQ2AutoSizeType::FindMember(Member);
+		if (!pMember)
+			return false;
+
+		switch ((AutoSizeMembers)pMember->ID) {
+			case Active:
+				Dest.Int = AS_Config.OptPC || AS_Config.OptNPC || AS_Config.OptPet || AS_Config.OptMerc || AS_Config.OptMount || AS_Config.OptCorpse || AS_Config.OptSelf;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case AutoSave:
+				Dest.Int = AS_Config.OptAutoSave;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizePC:
+				Dest.Int = AS_Config.OptPC;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizeNPC:
+				Dest.Int = AS_Config.OptNPC;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizePets:
+				Dest.Int = AS_Config.OptPet;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizeMercs:
+				Dest.Int = AS_Config.OptMerc;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizeMounts:
+				Dest.Int = AS_Config.OptMount;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizeCorpse:
+				Dest.Int = AS_Config.OptCorpse;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case ResizeSelf:
+				Dest.Int = AS_Config.OptSelf;
+				Dest.Type = datatypes::pBoolType;
+				return true;
+			case Range:
+				Dest.Int = AS_Config.ResizeRange;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizePC:
+				Dest.Int = AS_Config.SizePC;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizeNPC:
+				Dest.Int = AS_Config.SizeNPC;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizePets:
+				Dest.Int = AS_Config.SizePet;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizeMercs:
+				Dest.Int = AS_Config.SizeMerc;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizeMounts:
+				Dest.Int = AS_Config.SizeMount;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizeCorpse:
+				Dest.Int = AS_Config.SizeCorpse;
+				Dest.Type = datatypes::pIntType;
+				return true;
+			case SizeSelf:
+				Dest.Int = AS_Config.SizeSelf;
+				Dest.Type = datatypes::pIntType;
+				return true;
+		}
+		return false;
+	}
+
+	bool ToString(MQVarPtr VarPtr, char* Destination) {
+		return true;
```

Comment:
updated

Reply:
since it doesn't really do anything and behaves in a well defined manner. However we could also just not have this function at all. But this is fine, you don't need to change it, just want to point out the option.